src/Routes/Routes.tsx

```diff
@@ -56,6 +56,23 @@ export const Routes = () => {
     connectDemo,
   ])
 
+  /**
+   * Memoize the route list to avoid unnecessary cascading re-renders
+   * It should only re-render if the wallet changes
+   */
+  const privateRoutesList = useMemo(
+    () =>
+      appRoutes.map((route, index) => {
+        const MainComponent = route.main
+        return (
+          <PrivateRoute key={index} path={route.path} exact hasWallet={hasWallet}>
```

Comment:
```suggestion
          <PrivateRoute key={route.path} path={route.path} exact hasWallet={hasWallet}>
```

any chance we can use `route.path` as the `key`?

Reply:
@gomesalexandre to confirm we haven't bought back the native wallet label flicker

Reply:
Confirmed this causes a whole tree re-render, bringing back the flicker.


https://user-images.githubusercontent.com/17035424/188967585-188ff5a5-a506-4d40-a8ec-59e9160232af.mov

This is because of the inner workings of react, and by extension the implementation of react-router itself and how they interact with each other.

All `<Route />` components within a `<Switch />` would normally have no key when used as a regular JSX node, since we're not in the context of a JSX node iteration.

When mapped, things are a bit different. The `key` property is used by react to identify different instances of a component. When encountering a component with a different `key`, the reconciliation algorithm of react will run  unmount the current one, cleanup effects, and mount this new instance instead (https://kentcdodds.com/blog/understanding-reacts-key-prop).

Which *is* the expected behavior when mapping over "regular" JSX trees, but in the context of `<Route />` components, which uses the node of `<Switch />,  unmounting and remounting `<Route />` escapes from the logic of react-router, and effectively re-renders the whole tree of said route.

tl;dr use a unique key across all iterations of a `mappesRoutes.map()` call

Reply:
I resolved this by keeping @0xApotheosis ' change to use a static value.